[2024-06-14T11:41:15.634+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: simple_spark_dag.spark_submit_task scheduled__2024-06-13T11:40:51.220755+00:00 [queued]>
[2024-06-14T11:41:15.646+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: simple_spark_dag.spark_submit_task scheduled__2024-06-13T11:40:51.220755+00:00 [queued]>
[2024-06-14T11:41:15.648+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 1
[2024-06-14T11:41:15.668+0000] {taskinstance.py:1380} INFO - Executing <Task(SparkSubmitOperator): spark_submit_task> on 2024-06-13 11:40:51.220755+00:00
[2024-06-14T11:41:15.704+0000] {standard_task_runner.py:57} INFO - Started process 8692 to run task
[2024-06-14T11:41:15.720+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'simple_spark_dag', 'spark_submit_task', 'scheduled__2024-06-13T11:40:51.220755+00:00', '--job-id', '65', '--raw', '--subdir', 'DAGS_FOLDER/simple_spark_dag.py', '--cfg-path', '/tmp/tmpyp3lk137']
[2024-06-14T11:41:15.753+0000] {standard_task_runner.py:85} INFO - Job 65: Subtask spark_submit_task
[2024-06-14T11:41:16.112+0000] {task_command.py:415} INFO - Running <TaskInstance: simple_spark_dag.spark_submit_task scheduled__2024-06-13T11:40:51.220755+00:00 [running]> on host 963b3fbfa4c9
[2024-06-14T11:41:16.443+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='simple_spark_dag' AIRFLOW_CTX_TASK_ID='spark_submit_task' AIRFLOW_CTX_EXECUTION_DATE='2024-06-13T11:40:51.220755+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-06-13T11:40:51.220755+00:00'
[2024-06-14T11:41:16.496+0000] {base.py:73} INFO - Using connection ID 'Spark_connection' for task execution.
[2024-06-14T11:41:16.499+0000] {spark_submit.py:401} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --conf spark.hadoop.fs.s3a.endpoint=http://minio:9000 --conf spark.hadoop.fs.s3a.access.key=khaiVDT2024 --conf spark.hadoop.fs.s3a.secret.key=****** --conf spark.hadoop.fs.s3a.path.style.access=true --name spark_minio_job --deploy-mode client /opt/***/jobs/pyspark_test.py
[2024-06-14T11:41:16.794+0000] {spark_submit.py:571} INFO - /home/***/.local/lib/python3.9/site-packages/pyspark/bin/load-spark-env.sh: line 68: ps: command not found
[2024-06-14T11:41:29.237+0000] {spark_submit.py:571} INFO - python3: can't open file '/opt/***/jobs/pyspark_test.py': [Errno 2] No such file or directory
[2024-06-14T11:41:29.652+0000] {spark_submit.py:571} INFO - 24/06/14 11:41:29 INFO ShutdownHookManager: Shutdown hook called
[2024-06-14T11:41:29.670+0000] {spark_submit.py:571} INFO - 24/06/14 11:41:29 INFO ShutdownHookManager: Deleting directory /tmp/spark-8a258023-6431-40a0-8106-0cfdb333b882
[2024-06-14T11:41:30.634+0000] {taskinstance.py:1935} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 502, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.hadoop.fs.s3a.endpoint=http://minio:9000 --conf spark.hadoop.fs.s3a.access.key=khaiVDT2024 --conf spark.hadoop.fs.s3a.secret.key=****** --conf spark.hadoop.fs.s3a.path.style.access=true --name spark_minio_job --deploy-mode client /opt/***/jobs/pyspark_test.py. Error code is: 2.
[2024-06-14T11:41:30.795+0000] {taskinstance.py:1398} INFO - Marking task as FAILED. dag_id=simple_spark_dag, task_id=spark_submit_task, execution_date=20240613T114051, start_date=20240614T114115, end_date=20240614T114130
[2024-06-14T11:41:32.400+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 65 for task spark_submit_task (Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.hadoop.fs.s3a.endpoint=http://minio:9000 --conf spark.hadoop.fs.s3a.access.key=khaiVDT2024 --conf spark.hadoop.fs.s3a.secret.key=****** --conf spark.hadoop.fs.s3a.path.style.access=true --name spark_minio_job --deploy-mode client /opt/***/jobs/pyspark_test.py. Error code is: 2.; 8692)
[2024-06-14T11:41:32.752+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2024-06-14T11:41:33.947+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
